hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsMesosConstraints.java

```diff
@@ -0,0 +1,104 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.mesos.Protos.Attribute;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.Value.Range;
+
+/**
+ * HDFS Mesos offer constraints checker class implementation.
+ */
+public class HdfsMesosConstraints {
+
+  private final Map<String, String> constraints;
+
+  public HdfsMesosConstraints(Map<String, String> mesosSlaveConstraints) {
+    this.constraints = mesosSlaveConstraints;
+  }
```

Comment:
To simplify this code quite a bit, let's reset found to false at the beginning of our loop. Then let's not explicitly set it to false. We set it to true when necessary.  This will save us a bunch of code.